src/stats/stats_reader.cpp

```diff
@@ -960,7 +960,16 @@ void stats_reader_handler(sh_mem_t* p_sh_mem, int pid)
 	bpool_instance_block_t curr_bpool_blocks[NUM_OF_SUPPORTED_BPOOLS];
 	iomux_stats_t prev_iomux_blocks;
 	iomux_stats_t curr_iomux_blocks;
-	
+
+	prev_instance_blocks = (socket_instance_block_t*)alloca(p_sh_mem->max_skt_inst_num);
+	if (NULL == prev_instance_blocks) {
+		return ;
```

Comment:
If you look at GNU examples for alloca usage you notice that you don't ask about NULL as return value (since your stack memory cannot be NULL). If using malloc - then testing for NULL return value is appropriate.
http://www.gnu.org/software/libc/manual/html_node/Alloca-Example.html

Anyway, as requested above - change alloca to malloc or new